pandas/core/groupby/ops.py

```diff
@@ -646,20 +621,18 @@ def _aggregate(
         return result
 
     def _transform(
-        self,
-        result,
-        values,
-        comp_ids,
-        transform_func,
-        is_numeric,
-        is_datetimelike,
-        **kwargs
+        self, result, values, comp_ids, transform_func, is_datetimelike, **kwargs
     ):
 
         comp_ids, _, ngroups = self.group_info
         if values.ndim > 2:
             # punting for now
             raise NotImplementedError("number of dimensions is currently limited to 2")
+        elif transform_func is libgroupby.group_rank:
```

Comment:
May or may not be viable but is there a way to make `transform_func` a partial somewhere upstream to just freeze the one-off arguments that rank would need rather than special-casing here?

Reply:
I think for that we'd need to do a partial on all the other cases, or just add the ngroups kwarg to group_rank signature.

On the margin I'd like to have fewer partials/lambdas floating around the groupby code

Reply:
can you update for this (and for n_th above as well) so we don't have differeing signatures

Reply:
I agree with Jeff and think the unused args would be preferable, at least to try and make these as generic as possible. Could also add a check within the function bodies that they are unused.

I've been guilty of this in the past myself but I think adding special cases in methods like this for one-off function applications is more difficult to track over time

Reply:
so this is easy to do for group_rank, but would mean a small behavior change for group_nth (which currently ignores min_count).  do we want that?

Reply:
ok let's fix rank for now and discuss nth? (we can also fix the signature and just ignore the arg for now).